pkg/nodemodel/fileassets.go

```diff
@@ -42,22 +42,40 @@ type FileAssets struct {
 	// NodeUpAssets are the assets for downloading nodeup
 	NodeUpAssets map[architectures.Architecture]*assets.MirroredAsset
 
+	// AssetsSupportedKubelet are the assets for downloading nodeup, when the masters have not yet been rolled out to the version in the ClusterSpec
+	AssetsSupportedKubelet map[architectures.Architecture][]*assets.MirroredAsset
+
 	Cluster *kops.Cluster
 }
 
-// AddFileAssets adds the file assets within the assetBuilder
 func (c *FileAssets) AddFileAssets(assetBuilder *assets.AssetBuilder) error {
+	var err error
+	c.Assets, err = c.addFileAssets(assetBuilder, c.Cluster.Spec.KubernetesVersion)
+	if err != nil {
+		return err
+	}
+	if len(assetBuilder.KubeletSupportedVersion) > 0 && assetBuilder.KubeletSupportedVersion != c.Cluster.Spec.KubernetesVersion {
+		c.AssetsSupportedKubelet, err = c.addFileAssets(assetBuilder, assetBuilder.KubeletSupportedVersion)
```

Comment:
Ah - great point - we need different assets.  Although I'm starting to worry about complexity now...

Reply:
Yes, the complexity here worried me a bit as well as I dug a bit into how to achieve this. The code isn't perfect but i tried to do it without having to refactor many things, if you have any better ideas we can either improve this down the line or i can make any adjustments :)